climada_petals/entity/exposures/openstreetmap/osm_dataloader.py

```diff
@@ -0,0 +1,945 @@
+        Returns
+        -------
+        str : Geofabrik ownload-api for the requested country.
+        
+        See also
+        --------
+        DICT_GEOFABRIK for exceptions / special regions.
+        """
+
+        if file_format == 'shp':
+            return f'{self.geofabrik_url}{DICT_GEOFABRIK[iso3][0]}/{DICT_GEOFABRIK[iso3][1]}-latest-free.shp.zip'
+        elif file_format == 'pbf':
+            return f'{self.geofabrik_url}{DICT_GEOFABRIK[iso3][0]}/{DICT_GEOFABRIK[iso3][1]}-latest.osm.pbf'
+        else:
+            LOGGER.error('invalid file format. Please choose one of [shp, pbf]')
+
+    def get_data_geofabrik(self, iso3, file_format='pbf', save_path=DATA_DIR):
+        """
+        Download country files with all OSM map info from the provider 
+        Geofabrik.de, if doesn't exist, yet.
+        
+        Parameters
+        ----------
+        iso3 : str
+            ISO3 code of country to download
+            Exceptions: Russia is divided into European and Asian part 
+            ('RUS-E', 'RUS-A'), Canary Islands are 'IC'.
+        file_format : str
+            Format in which file should be downloaded; options are 
+            ESRI Shapefiles (shp), which can easily be loaded into gdfs,
+            or osm-Protocolbuffer Binary Format (pbf), which is smaller in 
+            size, but has a more complicated query syntax to load (functions
+            are provided in the OSMFileQuery class).
+        save_path : str or pathlib.Path
+            Folder in which to save the file
+            
+        Returns
+        -------
+        None
+            File is downloaded and stored unter save_path + filename
+        
+        See also
+        --------
+        DICT_GEOFABRIK for exceptions / special regions.
+        """
+        
+        download_url = self._create_gf_download_url(iso3, file_format)
+        local_filepath = Path(save_path , download_url.split('/')[-1])
+        if not Path(local_filepath).is_file():
+            LOGGER.info(f'Downloading file as {local_filepath}')
+            urllib.request.urlretrieve(download_url, local_filepath)
+        else:
+            LOGGER.info(f'file already exists as {local_filepath}')
+
+    def get_data_planet(self, 
+                        save_path=Path(DATA_DIR,'planet-latest.osm.pbf')):
+        """
+        Download the entire planet file from the OSM server (ca. 60 GB).
+        
+        Parameters
+        ----------
+        save_path : str or pathlib.Path
+        """
+        
+        if not Path(save_path).is_file():
+            LOGGER.info(f'Downloading file as {save_path}')
+            urllib.request.urlretrieve(self.planet_url, save_path)
+        else:
+            LOGGER.info(f'file already exists as {save_path}')
+
+    def _osmosis_extract(self, shape, path_planet, path_extract):
+        """
+        Runs the command line tool osmosis to cut out all map info within 
+        shape, from the osm planet file, unless file already exists.
```

Comment:
Is there a reason why you the default beviour isn't to overwrite an existing file? I would imagine that it's more convenient to be able to overwrite by default (especially if you're extracting to temporary files or files with arbitrary names) than to save people time who accidentally extract the same thing twice.

Maybe we need an `overwrite=False` parameter (or `overwrite=True`, which feels safer to me)

Reply:
osmosis_extract takes forever (!) to crop the desired area from the planet file. So you really don't want to mess up. I added a parameter `overwrite` that has a default of False in 243d11b4